source/_localCaptioner/captioner.py

```diff
@@ -0,0 +1,379 @@
+		"""Generate text using greedy search.
+
+		Args:
+			encoderHiddenStates: Encoder hidden states.
+			maxLength: Maximum generation length.
+
+		Returns:
+			Generated text string.
+		"""
+		if maxLength is None:
+			maxLength = self.maxLength
+
+		# Initialize input sequence
+		inputIds = np.array([[self.bosTokenId]], dtype=np.int64)
+		generatedTokens = []
+
+		# Initialize past_key_values
+		pastKeyValues = self._initializePastKeyValues(batchSize=1)
+
+		for step in range(maxLength):
+			# Prepare decoder inputs
+			decoderInputs = {
+				"input_ids": inputIds if step == 0 else np.array([[generatedTokens[-1]]], dtype=np.int64),
+				"encoder_hidden_states": encoderHiddenStates,
+				"use_cache_branch": np.array([1], dtype=np.bool_),
+			}
+
+			# Add past_key_values to inputs
+			decoderInputs.update(pastKeyValues)
+
+			# Run decoder
+			decoderOutputs = self.decoderSession.run(None, decoderInputs)
+			logits = decoderOutputs[0]  # Shape: (batch_size, seq_len, vocab_size)
+
+			# Greedy selection of next token
+			nextTokenLogits = logits[0, -1, :]  # Logits for last position
+			nextTokenId = int(np.argmax(nextTokenLogits))
+
+			# Check if generation should end
+			if nextTokenId == self.eosTokenId:
+				break
+
+			generatedTokens.append(nextTokenId)
+
+			# Update past_key_values from outputs
+			if len(decoderOutputs) > 1:
+				for layerIdx in range(self.nLayer):
+					if len(decoderOutputs) > 1 + layerIdx * 2 + 1:
+						pastKeyValues[f"past_key_values.{layerIdx}.key"] = decoderOutputs[1 + layerIdx * 2]
+						pastKeyValues[f"past_key_values.{layerIdx}.value"] = decoderOutputs[
+							1 + layerIdx * 2 + 1
+						]
+
+			# Avoid sequences that are too long
+			if len(generatedTokens) >= self.nCtx - 1:
+				break
+
+		# Decode generated text
+		return self.decodeTokens(generatedTokens)
+
+	def generateCaption(
+		self,
+		image: str | bytes,
+		maxLength: int | None = None,
+	) -> str:
+		"""Generate image caption.
+
+		Args:
+			image: Image file path or binary data.
+			maxLength: Maximum generation length.
+
+		Returns:
+			Generated image caption.
+		"""
+		# Preprocess image
+		imageArray = self.preprocessImage(image)
+
+		# Encode image
+		encoderHiddenStates = self.encodeImage(imageArray)
+
+		# Generate text
+		caption = self.generateWithGreedy(encoderHiddenStates, maxLength)
+
+		return caption
+
+
+def benchmarkInference(
+	captioner: ImageCaptioner,
+	imagePath: str,
+	numRuns: int = 5,
+) -> None:
+	"""Benchmark inference performance.
+
+	Args:
+		captioner: Model instance.
+		imagePath: Test image path.
+		numRuns: Number of runs for benchmarking.
+	"""
+	log.info(f"Running benchmark with {numRuns} iterations...")
+
+	# Warm up
+	captioner.generateCaption(imagePath)
+
+	# Test greedy search
+	startTime = time.time()
+	for _ in range(numRuns):
+		captioner.generateCaption(imagePath)
+	greedyTime = (time.time() - startTime) / numRuns
+
+	log.info("Average inference time:")
+	log.info(f"  Greedy search: {greedyTime:.3f}s")
```

Comment:
We generally don't want scripts like this as part of the API or committed to source. I'd suggest adding this to a manual test plan in `tests/manual`. i.e. add a manual test plan to test our captioning response time, and suggest using this in the NVDA python console.